akd_local_auditor/src/storage/mod.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under both the MIT license found in the
+// LICENSE-MIT file in the root directory of this source tree and the Apache
+// License, Version 2.0 found in the LICENSE-APACHE file in the root directory
+// of this source tree.
+
+//! This module holds the generic storage interaction layer trait along with the underlying implemented
+//! storage interactions
+
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use clap::Subcommand;
+use std::convert::TryFrom;
+
+// ************************************ Implementations ************************************ //
+
+pub mod dynamodb;
+pub mod s3;
+
+/// Storage options for retrieving audit proofs
+#[derive(Subcommand, Clone, Debug)]
+pub enum StorageSubcommand {
+    /// Amazon S3 compatible storage
+    S3(s3::S3ClapSettings),
+    // /// DynamoDB
+    // DynamoDb(dynamodb::DynamoDbClapSettings),
+}
+
+// ************************************ Trait and Type Definitions ************************************ //
+
+/// Represents the summary of an epoch, and a unique key referring to the raw object in native storage (if needed)
+#[derive(Clone)]
```

Comment:
Might as well [derive other common traits](https://rust-lang.github.io/api-guidelines/interoperability.html#c-common-traits) that might come in handy.